### PR TITLE
Strictly type everything in .github and tools

### PR DESCRIPTION
**Pull request by samestep**
This PR greatly simplifies `mypy-strict.ini` by strictly typing everything in `.github` and `tools`, rather than picking and choosing only specific files in those two dirs. It also removes `warn_unused_ignores` from `mypy-strict.ini`, for reasons described in https://github.com/pytorch/pytorch/pull/56402#issuecomment-822743795: basically, that setting makes life more difficult depending on what libraries you have installed locally vs in CI (e.g. `ruamel`).

**Test plan:**

```
flake8
mypy --config mypy-strict.ini
```

### CODE REVIEW

**Review by malfet — APPROVED** [review]
mypy is an authority here: if it says its green will merge it!

File: tools/autograd/gen_annotated_fn_args.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ def gen_annotated(native_yaml_path: str, out: str, autograd_dir: str) -> None:
 
     template_path = os.path.join(autograd_dir, 'templates')
     fm = FileManager(install_dir=out, template_dir=template_path, dry_run=False)
-    fm.write_with_template('annotated_fn_args.py', 'annotated_fn_args.py', lambda: {
+    fm.write_with_template('annotated_fn_args.py', 'annotated_fn_args.py.in', lambda: {
```
Comment (walterddr) [inline tools/autograd/gen_annotated_fn_args.py:51]: the build error is legit. you also need to change `caffe2/CMakeLists.txt`
```
--- a/caffe2/CMakeLists.txt
+++ b/caffe2/CMakeLists.txt
@@ -492,7 +492,7 @@ if(NOT INTERN_BUILD_MOBILE OR NOT BUILD_CAFFE2_MOBILE)
     "${TOOLS_PATH}/autograd/templates/python_linalg_functions.cpp"
     "${TOOLS_PATH}/autograd/templates/python_special_functions.cpp"
     "${TOOLS_PATH}/autograd/templates/variable_factories.h"
-    "${TOOLS_PATH}/autograd/templates/annotated_fn_args.py.in"
+    "${TOOLS_PATH}/autograd/templates/annotated_fn_args.py"
```

Comment (samestep) [reply tools/autograd/gen_annotated_fn_args.py:51]: trying that now, thanks for the catch!

**Review by walterddr — APPROVED** [review]
looks good. 

would suggest landing this during off peak hour to avoid cross merge :-P

**Pull request closed without merging**